
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://KU-GDSC.github.io/documentation/workflows/wgs/">
      
      
        <link rel="prev" href="../microbial_rnaseq/">
      
      
      
      <link rel="icon" href="../../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.5.2, mkdocs-material-9.3.1">
    
    
      
        <title>Whole-genome Sequencing Data Variant Calling - KU Genomic Data Science Core</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.046329b4.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.85d0ee34.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="slate" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#whole-genome-sequencing-data-variant-calling" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../.." title="KU Genomic Data Science Core" class="md-header__button md-logo" aria-label="KU Genomic Data Science Core" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            KU Genomic Data Science Core
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Whole-genome Sequencing Data Variant Calling
            
          </span>
        </div>
      </div>
    </div>
    
      
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/KU-GDSC/ku-gdsc.github.io/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    KU-GDSC/ku-gdsc.github.io
  </div>
</a>
      </div>
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../../.." class="md-tabs__link">
        
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../../../people/" class="md-tabs__link">
        
  
    
  
  People

      </a>
    </li>
  

      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../../../policies/" class="md-tabs__link">
        
  
    
  
  Policies

      </a>
    </li>
  

      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../../../services/" class="md-tabs__link">
        
  
    
  
  Services Offered

      </a>
    </li>
  

      
        
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../../" class="md-tabs__link">
          
  
    
  
  Documentation and Training

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


  

<nav class="md-nav md-nav--primary md-nav--lifted md-nav--integrated" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../.." title="KU Genomic Data Science Core" class="md-nav__button md-logo" aria-label="KU Genomic Data Science Core" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    KU Genomic Data Science Core
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/KU-GDSC/ku-gdsc.github.io/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    KU-GDSC/ku-gdsc.github.io
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
    <li class="md-nav__item">
      <a href="../../.." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
    <li class="md-nav__item">
      <a href="../../../people/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    People
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
    <li class="md-nav__item">
      <a href="../../../policies/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Policies
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
    <li class="md-nav__item">
      <a href="../../../services/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Services Offered
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
    
    
      
        
          
        
      
        
      
        
      
        
      
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" checked>
        
          
          
          <div class="md-nav__link md-nav__container">
            <a href="../../" class="md-nav__link ">
              
  
  <span class="md-ellipsis">
    Documentation and Training
  </span>
  

            </a>
            
              <label class="md-nav__link " for="__nav_5">
                <span class="md-nav__icon md-icon"></span>
              </label>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Documentation and Training
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../../quick_starts/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Quick Start Guides
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../../slow_starts/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Detailed Guides
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../../cheatsheet/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Generally useful *nix command cheatsheet
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
    
    
      
        
          
        
      
        
      
        
      
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_5" checked>
        
          
          
          <div class="md-nav__link md-nav__container">
            <a href="../" class="md-nav__link ">
              
  
  <span class="md-ellipsis">
    Workflows
  </span>
  

            </a>
            
              <label class="md-nav__link " for="__nav_5_5">
                <span class="md-nav__icon md-icon"></span>
              </label>
            
          </div>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_5_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_5_5">
            <span class="md-nav__icon md-icon"></span>
            Workflows
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../rnaseq/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    RNA-seq
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../microbial_rnaseq/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Microbial RNA-seq
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Whole-genome Sequencing Data Variant Calling
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Whole-genome Sequencing Data Variant Calling
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#summary-of-methods" class="md-nav__link">
    Summary of methods
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#parameters" class="md-nav__link">
    Parameters
  </a>
  
    <nav class="md-nav" aria-label="Parameters">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#required-parameters" class="md-nav__link">
    Required parameters:
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#optional-parameters" class="md-nav__link">
    Optional parameters:
  </a>
  
    <nav class="md-nav" aria-label="Optional parameters:">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#general-optional-parameters" class="md-nav__link">
    General optional parameters:
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fastp-filtering-parameters" class="md-nav__link">
    fastp filtering parameters:
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bwa-mem-mapping-parameters" class="md-nav__link">
    BWA mem mapping parameters:
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#variant-calling-parameters" class="md-nav__link">
    Variant calling parameters:
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#variant-filtering-parameters" class="md-nav__link">
    Variant filtering parameters:
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pipeline-default-outputs" class="md-nav__link">
    Pipeline Default Outputs
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


<h1 id="whole-genome-sequencing-data-variant-calling">Whole-genome Sequencing Data Variant Calling<a class="headerlink" href="#whole-genome-sequencing-data-variant-calling" title="Permanent link">#</a></h1>
<h2 id="summary-of-methods">Summary of methods<a class="headerlink" href="#summary-of-methods" title="Permanent link">#</a></h2>
<ul>
<li>Index generated with <a href="https://github.com/lh3/bwa" target="_blank">BWA</a></li>
<li>Filter and trim reads with <a href="https://github.com/OpenGene/fastp" target="_blank">fastp</a></li>
<li>Strand determination with <a href="https://pypi.org/project/how-are-we-stranded-here/" target="_blank">how are we stranded here</a></li>
<li>Quality checking of reads with <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/" target="_blank">FastQC</a></li>
<li>Reads mapped to reference <a href="https://github.com/lh3/bwa" target="_blank">BWA MEM</a></li>
<li>Alignment post-processing and QC with <a href="https://broadinstitute.github.io/picard/" target="_blank">Picard</a></li>
<li>Variant calling with <a href="https://samtools.github.io/bcftools/bcftools.html" target="_blank">bcftools</a></li>
<li>Variant filtering with <a href="https://gatk.broadinstitute.org" target="_blank">GATK</a></li>
<li>Aggregation of QC tables using <a href="https://multiqc.info" target="_blank">MultiQC</a></li>
</ul>
<h2 id="parameters">Parameters<a class="headerlink" href="#parameters" title="Permanent link">#</a></h2>
<h3 id="required-parameters">Required parameters:<a class="headerlink" href="#required-parameters" title="Permanent link">#</a></h3>
<ul>
<li>
<p><code>--pubdir</code> </p>
<ul>
<li>Default: <code>/&lt;PATH&gt;</code> </li>
<li>Description: The directory that the saved outputs will be stored.</li>
</ul>
</li>
<li>
<p><code>-w</code></p>
<ul>
<li>Default: <code>/&lt;PATH&gt;</code></li>
<li>The directory that all intermediary files and nextflow processes utilize. This directory can become quite large. This should be a location on scratch space or other directory with ample storage.</li>
</ul>
</li>
<li>
<p><code>--sample_folder</code> </p>
<ul>
<li>Default: <code>/&lt;PATH&gt;</code></li>
<li>The path to the folder that contains all the samples to be run by the pipeline. The files in this path can also be symbolic links.</li>
</ul>
</li>
<li>
<p><code>--fasta</code></p>
<ul>
<li>Default: <code>/&lt;PATH&gt;</code></li>
<li>Path to the reference genome in FASTA format</li>
</ul>
</li>
<li>
<p><code>--read_type</code></p>
<ul>
<li>Default: <code>PE</code></li>
<li>Comment: Type of reads: paired end (PE) or single end (SE).</li>
</ul>
</li>
</ul>
<h3 id="optional-parameters">Optional parameters:<a class="headerlink" href="#optional-parameters" title="Permanent link">#</a></h3>
<h4 id="general-optional-parameters">General optional parameters:<a class="headerlink" href="#general-optional-parameters" title="Permanent link">#</a></h4>
<ul>
<li>
<p><code>--concat_lanes</code></p>
<ul>
<li>Default: <code>false</code></li>
<li>Options: <code>false</code> and <code>true</code>. Default: <code>false</code>. If this boolean is specified, FASTQ files will be concatenated by sample. Used in cases where samples are divided across individual sequencing lanes.</li>
</ul>
</li>
<li>
<p><code>--extension</code></p>
<ul>
<li>Default: <code>.fastq.gz</code></li>
<li>Expected file extension for input read files, modify if files are not compressed or have a different form (e.g. ".fastq" or ".fq.gz")</li>
</ul>
</li>
<li>
<p><code>--pattern</code></p>
<ul>
<li>Default: <code>*_R{1,2}*</code></li>
<li>Expected R1/R2 matching pattern for paired-end read files in the <code>--sample_folder</code> path. In concert with <code>--extension</code> the default values will match files such as <code>sampleID.L001.R1.fastq.gz</code> and <code>sampleID.L001.R2.fastq.gz</code></li>
</ul>
</li>
<li>
<p><code>keep_intermediate</code></p>
<ul>
<li>Default: <code>false</code></li>
<li>If <code>true</code> workflow will output intermediate alignment files (unsorted BAMs, etc.)</li>
</ul>
</li>
<li>
<p><code>keep_reference</code></p>
<ul>
<li>Default: <code>false</code></li>
<li>If true workflow will save a copy of the BWA indices to the output directory</li>
</ul>
</li>
</ul>
<h4 id="fastp-filtering-parameters">fastp filtering parameters:<a class="headerlink" href="#fastp-filtering-parameters" title="Permanent link">#</a></h4>
<ul>
<li>
<p><code>--quality_phred</code></p>
<ul>
<li>Default: 15</li>
<li>Quality score threshold.</li>
</ul>
</li>
<li>
<p><code>--unqualified_perc</code></p>
<ul>
<li>Default: 40</li>
<li>Percent threshold of unqualified bases to pass reads.</li>
</ul>
</li>
<li>
<p><code>--tmpdir</code></p>
<ul>
<li>Default: "~/scratch/$USER/tmp/"</li>
</ul>
</li>
</ul>
<h4 id="bwa-mem-mapping-parameters">BWA mem mapping parameters:<a class="headerlink" href="#bwa-mem-mapping-parameters" title="Permanent link">#</a></h4>
<ul>
<li>
<p><code>--mismatch_penalty</code></p>
<ul>
<li>Default: "-B 8"</li>
</ul>
</li>
<li>
<p><code>--bwa_min_score</code></p>
<ul>
<li>Default: null</li>
<li>Threshold of alignment quality score to emit read alignment</li>
</ul>
</li>
</ul>
<h4 id="variant-calling-parameters">Variant calling parameters:<a class="headerlink" href="#variant-calling-parameters" title="Permanent link">#</a></h4>
<ul>
<li>
<p><code>--ploidy</code></p>
<ul>
<li>Default: 2</li>
<li>Options: 1, 2</li>
</ul>
</li>
<li>
<p><code>--mpileup_depth</code></p>
<ul>
<li>Default: 100</li>
<li>For a given position, read maximally INT reads per input file</li>
</ul>
</li>
<li>
<p><code>--skip_indels</code></p>
<ul>
<li>Default: false</li>
<li>If true, do not call indel sites</li>
</ul>
</li>
<li>
<p><code>--variants_only</code></p>
<ul>
<li>Default: true</li>
<li>If true, only emit variant sites (e.g. not homozygous reference)</li>
</ul>
</li>
</ul>
<h4 id="variant-filtering-parameters">Variant filtering parameters:<a class="headerlink" href="#variant-filtering-parameters" title="Permanent link">#</a></h4>
<ul>
<li>
<p><code>--filter_dp</code></p>
<ul>
<li>Default: "DP &lt; 25"</li>
<li>Sites with depth lower than this value will have a filter flag set as "LowCoverage"</li>
</ul>
</li>
<li>
<p><code>--filter_very_low_qual</code></p>
<ul>
<li>Default: "QUAL &lt; 30.0"</li>
<li>Sites with a quality score less than this value will have a filter flag set as "VeryLowQual"</li>
</ul>
</li>
<li>
<p><code>--filter_low_qual</code></p>
<ul>
<li>Default: "QUAL &gt; 30.0 &amp;&amp; QUAL &lt; 50.0"</li>
<li>Sites with a quality score less than this value will have a filter flag set as "LowQual"</li>
</ul>
</li>
<li>
<p><code>--filter_qd</code></p>
<ul>
<li>Default: "QD &lt; 1.5"</li>
<li>Sites with a depth-normalized quality score less than this value will have a filter flag set as "LowQD"</li>
</ul>
</li>
<li>
<p><code>--filter_fs</code></p>
<ul>
<li>Default: "FS &gt; 60.0"</li>
<li>Sites with a strand bias greater than this value will have a filter flag set as "StrandBias"</li>
</ul>
</li>
</ul>
<h2 id="pipeline-default-outputs">Pipeline Default Outputs<a class="headerlink" href="#pipeline-default-outputs" title="Permanent link">#</a></h2>
<table>
<thead>
<tr>
<th>Naming Convention</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>wgs_cohort_variants_bcftools_filtered.vcf</code></td>
<td>Soft-filtered VCF including all samples</td>
</tr>
<tr>
<td><code>wgs_cohort_variants_bcftools_filtered_genotype_table.tsv</code></td>
<td>RSEM-generated gene-level TPM counts merged across all samples</td>
</tr>
<tr>
<td><code>wgs_cohort_variants_bcftools_filtered_site_table.tsv</code></td>
<td>RSEM-generated isoform-level raw counts merged across all samples</td>
</tr>
<tr>
<td><code>wgs_report.html</code></td>
<td>Nextflow autogenerated report</td>
</tr>
<tr>
<td><code>index/</code></td>
<td>BWA-generated indices saved with optional parameter <code>keep_reference</code></td>
</tr>
<tr>
<td><code>multiqc/</code></td>
<td>MultiQC report summarizing quality metrics across all samples in the run</td>
</tr>
<tr>
<td><code>${sampleID}/stat/</code></td>
<td>QC and Picard metrics files for each sample</td>
</tr>
<tr>
<td><code>trace.txt</code></td>
<td>Nextflow trace of processes</td>
</tr>
</tbody>
</table>

  <hr>
<div class="md-source-file">
  <small>
    
      Last update:
      <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-date">January 12, 2025</span>
      
    
  </small>
</div>





                
              </article>
            </div>
          
          
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
            
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../../..", "features": ["navigation.path", "navigation.tabs", "navigation.tabs.sticky", "navigation.indexes", "navigation.top", "toc.integrate", "toc.follow"], "search": "../../../assets/javascripts/workers/search.dfff1995.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../../assets/javascripts/bundle.dff1b7c8.min.js"></script>
      
    
  </body>
</html>