# Microbial RNA-seq 

## Summary of methods 

* Index generated with [RSEM](http://deweylab.github.io/RSEM){:target="\_blank"}
* Filter and trim reads with [fastp](https://github.com/OpenGene/fastp){:target="\_blank"}
* Quality checking of reads with [FastQC](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/){:target="\_blank"}
* Reads mapped to reference and quantified with [RSEM](http://deweylab.github.io/RSEM){:target="\_blank"}
* Alignment post-processing and QC with [Picard](https://broadinstitute.github.io/picard/){:target="\_blank"}
* Aggregation of QC tables using [MultiQC](https://multiqc.info){:target="\_blank"}

## Parameters

### Required parameters:

* `--pubdir` 
	* Default: `/<PATH>` 
	* Description: The directory that the saved outputs will be stored.

* `-w`
    * Default: `/<PATH>`
    * The directory that all intermediary files and nextflow processes utilize. This directory can become quite large. This should be a location on scratch space or other directory with ample storage.

* `sample_folder` 
    * Default: `/<PATH>`
    * The path to the folder that contains all the samples to be run by the pipeline. The files in this path can also be symbolic links.

* `strandedness`
    * Default: `NA`
    * Supported options are `reverse_stranded`, `forward_stranded`, `non_stranded`

* `--fasta`
    * Default: `/<PATH>`
    * Path to the reference genome in FASTA format

* `--gff`
    * Default: `/<PATH>`
    * Path to the annotation for the reference genome in GFF3 format

* `--read_type`
    * Default: `PE`
    * Comment: Type of reads: paired end (PE) or single end (SE).

### fastp filtering paramenters:

* `--quality_phred`
    * Default: 15
    * Quality score threshold.

* `--unqualified_perc`
    * Default: 40
    * Percent threhold of unqualified bases to pass reads.


## Pipeline Default Outputs 
 
| Naming Convention                                                    | Description                                                                                                   |
| -------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------- |
| `microbial_rnaseq_report.html`                                        | Nextflow autogenerated report                                                                                 |
| `trace.txt`                                                          | Nextflow trace of processes                                                                                   |
| `multiqc/`                                                           | MultiQC report summarizing quality metrics across all samples in the run                                      |
| `${sampleID}/bam/${sampleID}.genome.bam`                             | RSEM-generated alignment of reads to the reference genome                                                     |
| `${sampleID}/bam/${sampleID}.transcript.bam`                         | RSEM-generated alignment of reads to the reference transcriptome                                              |
| `${sampleID}/${sampleID}.genes.results`                          | RSEM-generated quantification of gene-level count abundances                                                  |
| `${sampleID}/${sampleID}.isoforms.results`                       | RSEM-generated quantification of transcript-level count abundances                                            |
